beep/featurize.py

```diff
@@ -128,44 +130,46 @@ def init_full_model(cls, processed_cycler_run, init_pred_cycle=10, mid_pred_cycl
         Returns:
             beep.featurize.DegradationPredictor: DegradationPredictor corresponding to the ProcessedCyclerRun file.
         """
+        required_cycle_count = 100
+        if len(processed_cycler_data['summary']['discharge_capacity']) < required_cycle_count:
+            return None
+
         assert mid_pred_cycle > 10, 'Insufficient cycles for analysis'
         assert final_pred_cycle > mid_pred_cycle, 'Must have final_pred_cycle > mid_pred_cycle'
         ifinal = final_pred_cycle - 1  # python indexing
         imid = mid_pred_cycle - 1
-        iini = init_pred_cycle - 1
-        summary = processed_cycler_run.summary
-        assert len(processed_cycler_run.summary) > final_pred_cycle, 'cycle count must exceed final_pred_cycle'
+        assert len(processed_cycler_data['summary']['discharge_capacity']) > final_pred_cycle, 'cycle count must exceed final_pred_cycle'
         cycles_to_average_over = 40  # For nominal capacity, use median discharge capacity of first n cycles
 
-        interpolated_df = processed_cycler_run.cycles_interpolated
         X = pd.DataFrame(np.zeros((1, 20)))
         labels = []
         # Discharge capacity, cycle 2 = Q(n=2)
-        X[0] = summary.discharge_capacity[1]
+        X[0] = processed_cycler_data['summary']['discharge_capacity'][1]
         labels.append("discharge_capacity_cycle_2")
 
         # Max discharge capacity - discharge capacity, cycle 2 = max_n(Q(n)) - Q(n=2)
-        X[1] = max(summary.discharge_capacity[np.arange(final_pred_cycle)] - summary.discharge_capacity[1])
+        X[1] = max([x - processed_cycler_data['summary']['discharge_capacity'][1] for x in processed_cycler_data['summary']['discharge_capacity'][:final_pred_cycle]])
```

Comment:
This may be faster with something like
processed_cycler_data['summary']['discharge_capacity'][:final_pred_cycle].max() - processed_cycler_data['summary']['discharge_capacity'][1]